### ISSUE TITLE
how to get values of latent varaibles[Docs]

### ISSUE THREAD

**Issue by irum** [cornellius-gp/gpytorch#1229]
I would like to ask which function of likelihood gives the values for latent variable. 
The logits values of ` output = likelihood(feat)`  from softmax_likelihood.py gives values of latent variable or I modified a bit that file by adding the following line ` res.mixed_fs = mixed_fs` and get different values so I am confused which values are the values of latent variables.
 
the values for logits are:

            tensor([[[-1.0490e-05, -1.4917e+01, -2.1658e+01, -1.4043e+01, -1.1869e+01,
          -1.8907e+01, -1.3785e+01, -1.7071e+01, -1.3397e+01, -2.1994e+01]],

        [[ 0.0000e+00, -1.5744e+01, -2.7024e+01, -1.7415e+01, -1.4473e+01,
          -2.0655e+01, -1.5960e+01, -2.4946e+01, -1.6375e+01, -2.7051e+01]],

        [[-7.8773e-04, -8.8004e+00, -1.6850e+01, -1.2870e+01, -8.1776e+00,
          -1.6647e+01, -7.9869e+00, -1.1915e+01, -1.1893e+01, -1.7440e+01]],

        [[-1.0109e-04, -1.1599e+01, -1.7041e+01, -1.2888e+01, -9.3272e+00,
          -1.7727e+01, -1.5606e+01, -1.8522e+01, -1.9932e+01, -1.7542e+01]],

        [[ 0.0000e+00, -1.8660e+01, -2.0071e+01, -1.9172e+01, -1.4311e+01,
          -1.9895e+01, -2.0940e+01, -1.8697e+01, -1.5652e+01, -2.1551e+01]],

        [[-9.2506e-05, -9.3707e+00, -2.1120e+01, -1.4079e+01, -1.2300e+01,
          -1.6173e+01, -1.5570e+01, -1.8869e+01, -1.3365e+01, -1.8712e+01]],

        [[-3.4904e-04, -9.2426e+00, -1.9480e+01, -1.1088e+01, -8.3581e+00,
          -1.6245e+01, -1.3613e+01, -1.6262e+01, -1.3903e+01, -1.7658e+01]],

        [[-1.4215e-02, -1.0255e+01, -1.0759e+01, -1.6562e+01, -1.0963e+01,
          -1.1940e+01, -4.2699e+00, -1.2661e+01, -9.9330e+00, -1.4419e+01]],

        [[ 0.0000e+00, -1.4698e+01, -2.0045e+01, -1.9000e+01, -1.8215e+01,
          -2.2857e+01, -2.0540e+01, -2.6250e+01, -2.2410e+01, -2.3419e+01]],

        [[-2.2936e-03, -6.1106e+00, -2.0170e+01, -1.6071e+01, -9.5487e+00,
          -1.5588e+01, -1.5092e+01, -1.7855e+01, -1.4920e+01, -2.3176e+01]],

The values of modified line in softmax_likelihood.py are:

          tensor([[[ 1.4788e+01, -1.2951e-01, -6.8707e+00,  7.4444e-01,  2.9193e+00,
          -4.1189e+00,  1.0032e+00, -2.2833e+00,  1.3905e+00, -7.2061e+00]],

        [[ 1.7856e+01,  2.1122e+00, -9.1674e+00,  4.4139e-01,  3.3832e+00,
          -2.7984e+00,  1.8958e+00, -7.0900e+00,  1.4807e+00, -9.1949e+00]],

        [[ 1.1327e+01,  2.5270e+00, -5.5225e+00, -1.5431e+00,  3.1498e+00,
          -5.3198e+00,  3.3405e+00, -5.8789e-01, -5.6555e-01, -6.1123e+00]],

        [[ 1.3867e+01,  2.2677e+00, -3.1747e+00,  9.7842e-01,  4.5395e+00,
          -3.8604e+00, -1.7393e+00, -4.6555e+00, -6.0653e+00, -3.6752e+00]],

        [[ 1.6976e+01, -1.6845e+00, -3.0956e+00, -2.1966e+00,  2.6648e+00,
          -2.9192e+00, -3.9640e+00, -1.7210e+00,  1.3238e+00, -4.5752e+00]],

        [[ 1.3820e+01,  4.4492e+00, -7.3003e+00, -2.5870e-01,  1.5203e+00,
          -2.3536e+00, -1.7497e+00, -5.0492e+00,  4.5480e-01, -4.8923e+00]],

        [[ 1.2532e+01,  3.2902e+00, -6.9471e+00,  1.4449e+00,  4.1747e+00,
          -3.7122e+00, -1.0805e+00, -3.7289e+00, -1.3699e+00, -5.1253e+00]],

        [[ 1.0352e+01,  1.1056e-01, -3.9294e-01, -6.1963e+00, -5.9682e-01,
          -1.5738e+00,  6.0960e+00, -2.2955e+00,  4.3292e-01, -4.0528e+00]],

        [[ 1.8716e+01,  4.0180e+00, -1.3289e+00, -2.8361e-01,  5.0114e-01,
          -4.1407e+00, -1.8241e+00, -7.5340e+00, -3.6937e+00, -4.7024e+00]],

        [[ 1.3693e+01,  7.5851e+00, -6.4743e+00, -2.3753e+00,  4.1470e+00,
          -1.8924e+00, -1.3966e+00, -4.1591e+00, -1.2243e+00, -9.4804e+00]],

Q2 ).In GPflow library `predict_f` gives values for latent varaibles . What is the equivalent function in GPytorch.

Q3).   Could you please explain what is the meaning of the following code line in file categorical.py

` self.logits = logits - logits.logsumexp(dim=-1, keepdim=True)`

**Comment by jacobrgardner** [comment]
Q1) I didn't see a Q1 -- was there one?

Q2) For all GP models in GPyTorch, `likelihood(model(x))` is analogous to `predict_y`, while `model(x)` by itself is analogous to `predict_f`. A good way to think of this is that `likelihood(p(f|D))` means "solve/approximate `\int p(y|f)p(f|D)df`.

Q3) When dealing with probabilities, it is common to use log space. This is because if a_i is a very negative number representing a log probability, than the corresponding probability `exp(a_i)` will probably get rounded to 0. If `a_1_,...,a_n` represent unnormalized values in log space, then `b = a - a.logsumexp()` represents a normalized version of them, i.e. so that `\sum_i exp(b_i) = 1`.

**Comment by irum** [comment]
@jacobrgardner 
Thank you for your reply . 
Q1 was "I would like to ask which function of likelihood gives the values for latent variable." 
If you refer to the example [here](https://docs.gpytorch.ai/en/v1.1.1/examples/06_PyTorch_NN_Integration_DKL/Deep_Kernel_Learning_DenseNet_CIFAR_Tutorial.html) then `model(x) ` is analogous to `predict_f` ( which means giving you the values for latent varaibles)
